server/rest/src/main/java/org/infinispan/rest/Http11To2UpgradeHandler.java

```diff
@@ -61,6 +75,7 @@ private void configureHttp2(ChannelPipeline pipeline) {
    private void configureHttp1(ChannelPipeline pipeline) {
       final HttpServerCodec httpCodec = new HttpServerCodec(MAX_INITIAL_LINE_SIZE, MAX_HEADER_SIZE, restServer.getConfiguration().maxContentLength());
       pipeline.addLast(httpCodec);
+      pipeline.addLast(new CorsHandler(corsConfig));
```

Comment:
Is there any reason why we can't inline it here? I believe we don't need to reference.

And could you please confirm if it works with HTTP/2?

Reply:
Inline what exactly?

Regarding HTTP/2, I have no idea. AFAICT, ```XMLHttpRequest``` uses HTTP/1.x, in which circumstances they use HTTP/2 so that I can test?

Reply:
In any case, added a task to https://issues.jboss.org/browse/ISPN-8915 to investigate that.

Reply:
> inline what exactly?

Something like this:

```
pipeline.addLast(CorsConfigBuilder.forAnyOrigin().allowedRequestMethods(GET, POST, PUT, DELETE, HEAD, OPTIONS).build());
```

> Regarding HTTP/2, I have no idea. AFAICT, XMLHttpRequest uses HTTP/1.x, in which circumstances they use HTTP/2 so that I can test?

Is this feature connected only to `XMLHttpRequest`? If so, does it always use `HTTP/1.x`? I briefly looked through the [Wiki page](https://en.wikipedia.org/wiki/XMLHttpRequest) and couldn't find any information that this is `HTTP/1.x` specific...

Reply:
The ```corsConfig``` needs to be created once, inlining it would cause unnecessary allocations every time a channel is created.